android/src/main/java/net/infoxication/reactstarprnt/RNStarPrntModule.java

```diff
@@ -572,23 +580,23 @@ else if (command.hasKey("appendBytes")) {
                 }else builder.appendQrCode(command.getString("appendQrCode").getBytes(encoding), qrCodeModel, qrCodeLevel, cell);
             } else if (command.hasKey("appendBitmap")){
                 ContentResolver contentResolver = context.getContentResolver();
-                String uriString = command.getString("appendBitmap");
+                String _uriString = command.getString("appendBitmap");
+                String uriString =  _uriString.substring(_uriString.indexOf(",")  + 1);
                 boolean diffusion = (command.hasKey("diffusion")) ? command.getBoolean("diffusion") : true;
                 int width = (command.hasKey("width")) ? command.getInt("width") : 576;
                 boolean bothScale = (command.hasKey("bothScale")) ? command.getBoolean("bothScale") : true;
                 ICommandBuilder.BitmapConverterRotation rotation = (command.hasKey("rotation")) ? getConverterRotation(command.getString("rotation")) : getConverterRotation("Normal");
-                try {
-                    Uri imageUri =  Uri.parse(uriString);
-                    Bitmap bitmap = MediaStore.Images.Media.getBitmap(contentResolver, imageUri);
-                    if(command.hasKey("absolutePosition")){
-                        int position =  command.getInt("absolutePosition");
-                        builder.appendBitmapWithAbsolutePosition(bitmap, diffusion, width, bothScale, rotation, position);
-                    }else if(command.hasKey("alignment")){
-                        ICommandBuilder.AlignmentPosition alignmentPosition = getAlignment(command.getString("alignment"));
-                        builder.appendBitmapWithAlignment(bitmap, diffusion, width, bothScale, rotation, alignmentPosition);
-                    }else builder.appendBitmap(bitmap, diffusion, width, bothScale, rotation);
-                } catch (IOException e) {
 
+                final byte[] decodedBytes = Base64.decode(uriString, Base64.DEFAULT);
+                Bitmap bitmap = BitmapFactory.decodeByteArray(decodedBytes, 0, decodedBytes.length);
```

Comment:
I know this came from the other PR, but I'm wondering if dropping that try/catch completely was a good move

Reply:
ah yeah, probably not. I've added it back. along with removing a couple of imports that I noticed we previously removed.